linera-views/src/backends/scylla_db.rs

```diff
@@ -337,43 +375,69 @@ impl ScyllaDbClient {
         root_key: &[u8],
         batch: UnorderedBatch,
     ) -> Result<(), ScyllaDbStoreInternalError> {
-        let session = &self.session;
-        let mut batch_query = scylla::statement::batch::Batch::new(BatchType::Unlogged);
-        let mut batch_values = Vec::new();
-        let query1 = &self.write_batch_delete_prefix_unbounded;
-        let query2 = &self.write_batch_delete_prefix_bounded;
         Self::check_batch_len(&batch)?;
+        let session = &self.session;
+        let root_key = root_key.to_vec();
+
+        let mut futures: Vec<BoxFuture<'_, Result<(), ScyllaDbStoreInternalError>>> = Vec::new();
         for key_prefix in batch.key_prefix_deletions {
             Self::check_key_size(&key_prefix)?;
             match get_upper_bound_option(&key_prefix) {
                 None => {
-                    let values = vec![root_key.to_vec(), key_prefix];
-                    batch_values.push(values);
-                    batch_query.append_statement(query1.clone());
+                    let prepared_statement = &self.write_batch_delete_prefix_unbounded;
+                    let values = (root_key.clone(), key_prefix);
+                    futures.push(Box::pin(async move {
+                        session
+                            .execute_single_page(prepared_statement, values, PagingState::start())
+                            .await
+                            .map(|_| ())
+                            .map_err(Into::into)
+                    }));
                 }
-                Some(upper_bound) => {
-                    let values = vec![root_key.to_vec(), key_prefix, upper_bound];
-                    batch_values.push(values);
-                    batch_query.append_statement(query2.clone());
+                Some(upper) => {
+                    let prepared_statement = &self.write_batch_delete_prefix_bounded;
+                    let values = (root_key.clone(), key_prefix, upper);
+                    futures.push(Box::pin(async move {
+                        session
+                            .execute_single_page(prepared_statement, values, PagingState::start())
+                            .await
+                            .map(|_| ())
+                            .map_err(Into::into)
+                    }));
                 }
             }
         }
-        let query3 = &self.write_batch_deletion;
+        try_join_all(futures).await?;
+
+        let mut futures: Vec<BoxFuture<'_, Result<(), ScyllaDbStoreInternalError>>> = Vec::new();
```

Comment:
I guess the deletions of individual keys could even happen concurrently with the prefix deletions?